app/src/main/java/org/mozilla/fenix/ext/String.kt

```diff
@@ -28,21 +30,6 @@ fun String?.getHostFromUrl(): String? = try {
     null
 }
 
-fun String?.urlToTrimmedHost(): String {
-    return try {
-        val url = URL(this)
-        val firstIndex = url.host.indexOfFirst { it == '.' } + 1
-        val lastIndex = url.host.indexOfLast { it == '.' }
-
-        // Trim all but the title of the website from the hostname. 'www.mozilla.org' becomes 'mozilla'
-        when {
-            firstIndex - 1 == lastIndex -> url.host.substring(0, lastIndex)
-            firstIndex < lastIndex -> url.host.substring(firstIndex, lastIndex)
-            else -> url.host
-        }
-    } catch (e: MalformedURLException) {
-        this.getHostFromUrl() ?: ""
-    } catch (e: StringIndexOutOfBoundsException) {
-        this.getHostFromUrl() ?: ""
-    }
+suspend fun String.urlToTrimmedHost(context: Context): String {
```

Comment:
Is this an appropriate way to handle this method, @pocmo? I tried to pattern match what you implemented for `URLRenderer` in A-C.

Reply:
(1) Turn `PublicSuffixList` into a Singleton (like the other components that you can reach from a `Context` object, e.g. `SessionManager`). It is a pretty heavy thing that needs to load a bunch of things from disk. You don't want to do that multiple times and you definitely don't want to have multiple of those hanging around.

(2) Regarding the prefix: There used to be a method in Fennec, maybe Focus and should be in AC (but I can't find it.. if it doesn't exist we should create it eventually) that removes common prefixes, e.g. not only "www", but also "m" for mobile pages.

(3) Consider the edge cases: Not everything is a http(s) URL. What if the URL has no host? What about IP addresses? I'm not sure whether `PublicSuffixList` can deal with all those cases since it expects you to pass in a host and may not handle wonky values well.

Reply:
Found the prefix stripping in Focus: https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/utils/UrlUtils.java#L142

Reply:
Regarding 3) I tested with IP addresses and PublicSuffixList seems to handle it fine (the whole string is just returned e.g. `192.168.1.254`

Reply:
A-C issue: https://github.com/mozilla-mobile/android-components/issues/3299